python/kubeflow/trainer/utils/utils.py

```diff
@@ -107,6 +108,74 @@ def get_runtime_trainer_container(
     return None
 
 
+def _detect_trainer_from_image_patterns(image_name: str) -> Optional[types.Trainer]:
```

Comment:
@tenzen-y @Electronic-Waste @astefanutti @jskswamy @eoinfennessy @franciscojavierarceo Do we see any concerns with regex approach ?  It might be a good and simple method to start with, but I can imagine use cases where it wouldn't work. For example, users might have two DeepSpeed runtimes:

- One uses `torchrun` 
- Another uses `mpirun`.

Perhaps in the future we can support such scenarios.